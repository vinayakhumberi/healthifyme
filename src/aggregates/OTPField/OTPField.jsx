import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { Container, MessageSpace } from "./style";
import Text from '../../components/Text/Text';
import TextInput from '../../components/TextInput/TextInput';
import {
  OTP_REGEX
} from '../../constants/general.constants';

const OTPField = function (props) {
  const [showMessage, setShowMessage] = useState('');
  const [valid, setValid] = useState(false);
  const otpRef = useRef('');
  useEffect(() => {
    props.updateOTP({
      value: otpRef.current,
      status: valid
    })
  }, [valid]);
  const handleTextChange = (e) => {
    const data = e.target.value;
    if (showMessage) setShowMessage("");
    if (valid) setValid(false);
    otpRef.current = data;
    verifyOTP(data, false);
  };
  const handleVerify = (e) => {
    const data = e.target.value;
    verifyOTP(data, true);
  };
  const verifyOTP = (data, error) => {
    if (data.length === 0 && error) {
      setShowMessage("This field is mandatory");
    } else if (data.length !== 6 && error) {
      setShowMessage("Should be 6 characters long");
    } else {
      if (OTP_REGEX.test(data)) {
        setValid(true);
      } else {
        if (error) setShowMessage("Invalid OTP");
      }
    }
  };
  return (
    <Container
      data-test="otp-field-container"
    >
      <TextInput
        elementType="text"
        marginTop={10}
        marginBottom={0}
        paddingTop={10}
        paddingBottom={10}
        paddingLeft={10}
        paddingRight={10}
        width="100%"
        placeHolder="Enter your OTP"
        onChange={handleTextChange}
        onBlur={handleVerify}
      />
      <MessageSpace>
        {showMessage && <Text
          text={showMessage}
          color="#ec445c"
          paddingTop={5}
          paddingLeft={5}
        />}
      </MessageSpace>
    </Container>
  );
};
OTPField.propTypes = {
};

OTPField.defaultProps = {
};

export default OTPField;
