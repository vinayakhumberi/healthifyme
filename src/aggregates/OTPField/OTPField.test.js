import React from 'react';
import { mount, shallow, render } from 'enzyme';
import OTPField from './OTPField';

describe('<OTPField />', () => {
  let wrapper;
  const updateOTP = () => { };
  beforeEach(() => wrapper = shallow(<OTPField updateOTP={updateOTP} />));

  it('Should Render a OTPField Container', () => {
    expect(wrapper.find('[data-test="otp-field-container"]').length).toEqual(1);
  });
});


describe('<OTPField /> error check', () => {
  let wrapper;
  const updateOTP = () => { };
  beforeEach(() => wrapper = mount(<OTPField updateOTP={updateOTP} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('Should Render a OTPField Container', () => {
  //   expect(wrapper.find('[data-test="otp-field-container"]').length).toEqual(1);
  // });
  it('Should pass updateOTP prop to TextInput', () => {
    const input = wrapper.find('input');
    input.simulate('focus');
    input.simulate('change', { target: { value: 'Test Data' } });
    // expect(input.get(0).value).toEqual('Test Data');
  });
  it('Should show Error message is text is not more than 6 character long', () => {

  });
  it('Should show Error message if onBlur is fired and text is empty', () => {

  });
  it('Should show Error message if it is an invalid OTP', () => {

  });
});
