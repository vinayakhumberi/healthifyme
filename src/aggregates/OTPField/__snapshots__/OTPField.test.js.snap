// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OTPField /> error check renders correctly 1`] = `ReactWrapper {}`;
